src/Microsoft.ML.Data/Transforms/KeyToVector.cs

```diff
@@ -735,13 +735,13 @@ internal static class Defaults
             public const bool Bag = false;
         }
 
-        public KeyToVectorMappingEstimator(IHostEnvironment env, params KeyToVectorMappingTransformer.ColumnInfo[] columns)
+        internal KeyToVectorMappingEstimator(IHostEnvironment env, params KeyToVectorMappingTransformer.ColumnInfo[] columns)
             : this(env, new KeyToVectorMappingTransformer(env, columns))
         {
         }
 
-        public KeyToVectorMappingEstimator(IHostEnvironment env, string inputColumn, string outputColumn = null, bool bag = Defaults.Bag)
-            : this(env, new KeyToVectorMappingTransformer(env, new KeyToVectorMappingTransformer.ColumnInfo(inputColumn, outputColumn ?? inputColumn, bag)))
+        internal KeyToVectorMappingEstimator(IHostEnvironment env, string outputColumn, string inputColumn, bool bag = Defaults.Bag)
```

Comment:
>outputColumn, string inputColumn [](start = 74, length = 32)

Thank you @sfilipi! It was in a note somewhat down on the issue, but in a reply to @@justinormont where we were talking about consistency with existing argument names, I also observed that the names "inputs" and "outputs" are incorrect (since the inputs to outputs to estimators/transforms are *not* columns, but `IDataView`s), and should retain something closer the existing names we have for this `name` and `source`. Indeed, I think the bad naming was part of the reason some people got so badly confused in the first place.

Reply:
But these ones are `InputColumns` and `OutputColumns` and estimator operate over columns in `IDataView`. What is wrong with them? And why `name` and `source` and not `destination` and `source`? `name` is such a broad term to use it.

---
In reply to: [246974093](https://github.com/dotnet/machinelearning/pull/2118#discussion_r246974093) [](ancestors = 246974093)

Reply:
I hope i am not annoying and further fueling the conversation, but sharing my personal opinion:

1-  I agree that the order of arguments should be swapped, and have destination, than source. 
2- I find the name "name" for the output column confusing too. Yes it does make sense in the scenario, but you have to blink about 4 times before it does so; just because, as Ivan says, it is such a generic thing. 
The names InputColumn, and OutputColumn make a lot more sense to me too, since they are actually column names..  

The end!

---
In reply to: [246974845](https://github.com/dotnet/machinelearning/pull/2118#discussion_r246974845) [](ancestors = 246974845,246974093)

Reply:
The term `name` and `source` is a well established name in this codebase. The concept of the input and output for a *transform* is already being established as being the dataviews themselves. Transforms do *not* take columns as inputs and outputs, they take *data views* as inputs and outputs -- that's obvious from the interface specification. So: let's jsut agree, the dataviews are the inputs and outputs, full stop. In the case where a transform *happens* to be producing an additional column, we have parameterized the transform as producing a column with that "name." And when that parameterization also happens to pay attention to one column, we often call that a "source." Always.

It's interesting that you have a different interpretation, and I'd like to emphasize that I don't discount it, but it is distinct from how we've architected the old `IDataTransform` and I don't like the new interpretation for the reasons already discussed at length.

Going forward, I think this shows how dangerous it is to introduce a new interpretation without discussing that new interpretation with anyone, discussing its merits, etc. I'm sorry that was done, but not sorry that we're fixing it.

---
In reply to: [247204046](https://github.com/dotnet/machinelearning/pull/2118#discussion_r247204046) [](ancestors = 247204046,246974845,246974093)

Reply:
> Transforms do not take columns as inputs and outputs, they take data views as inputs and outputs 

Estimators during it construction doesn't take any dataview, they take columns which expected to be in dataview. We don't touch dataview till we call Fit. We do schema propagation for `GetOutputSchema` on `SchemaShape.Column`, 
Only thing which take data views as inputs as far as I know is `AppendDataViewRows` transform.

>The term name and source is a well established name in this codebase.

Internally, yes, mostly as legacy for command line. Legacy doesn't mean it's a great and useful for another task (Public Api).

> let's jsut agree, the dataviews are the inputs and outputs, full stop. 

I'm as user, don't see input columns and output columns in estimators as input and output dataviews. I think @sfilipi has the same opinion.  I can't even come up with estimator/transformer which would produce multiple dataviews (maybe evaluator with metrics as dataviews)

So far we have two people who think `name` and `source` is not a great idea (especially `name`) 
We do it not because we want to make you mad, we do it because we also care about project. 

>without discussing that new interpretation with anyone 

Can we actually have discussion regarding `name` and `source` vs `inputColumn` and `outputColumn` ? I would love to hear other people opinion, don't you?